deploy/helm/secret-operator/templates/secret_migration_job.yaml

```diff
@@ -0,0 +1,54 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "operator.fullname" . }}-secret-migration
+  labels:
+    {{- include "operator.labels" . | nindent 4 }}
+spec:
+  template:
+    metadata:
+      annotations:
+        "helm.sh/hook": pre-install
+        "helm.sh/hook-delete-policy": hook-succeeded
+      {{- with .Values.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "operator.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "operator.fullname" . }}-serviceaccount
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+      - name: migrate-secret
+        image: "{{ .Values.secretMigrationJob.image.repository }}:1.0.0-stackable{{ .Chart.AppVersion }}"
+        imagePullPolicy: {{ .Values.secretMigrationJob.image.pullPolicy }}
+        resources:
+            {{ .Values.secretMigrationJob.resources | toYaml | nindent 12 }}
+        command: ["bash", "-c"]
+        args:
+        - |
+          set -uo pipefail
+          SOURCE_NAMESPACE=default
+          TARGET_NAMESPACE={{ .Values.secretClasses.tls.caSecretNamespace | default .Release.Namespace }}
+
+          kubectl get secret -n $SOURCE_NAMESPACE secret-provisioner-tls-ca
+          exit_status=$?
+          # only continue if secret exists
+          if [ $exit_status -eq 0 ]; then
+            echo "secret exists in namespace $SOURCE_NAMESPACE"
+            kubectl get secret -n $TARGET_NAMESPACE secret-provisioner-tls-ca
+            exit_status=$?
+            # only continue if secret in target namespace does NOT exist
+            if [ $exit_status -eq 1 ]; then
+              echo "secret does not exist in namespace $TARGET_NAMESPACE"
+              # copy secret from default to {{ .Values.secretClasses.tls.caSecretNamespace | default .Release.Namespace }}
+              kubectl get secret -n $SOURCE_NAMESPACE secret-provisioner-tls-ca -o json | jq 'del(.metadata["namespace","creationTimestamp","resourceVersion","selfLink","uid"])' | kubectl apply -n $TARGET_NAMESPACE -f -
+            fi
+          fi
+      restartPolicy: Never
```

Comment:
@Techassi, curious why pre-commit skipped all the checks (incl trailing newline). Any ideas?

![image](https://github.com/user-attachments/assets/9f3e1c10-d1e2-494f-a4dd-86b0cb964d69).

It is not set to ignore in: https://github.com/stackabletech/secret-operator/blob/main/.pre-commit-config.yaml

Reply:
Generally, pre-commit will only trigger for modified files. For a full run you need to add `--all-files`.

Reply:
The yaml file was updated, so it should have triggered, no?
I would at least expect these hooks to run:
- trim trailing whitespace
- fix end of files
- detect aws credentials
- detect private keys
- yamllint

Reply:
Yes, I agree that at least _some_ of the checks (likely the ones @NickLarsenNZ mentioned) should have run.

Reply:
I feel like the detection of changed files works differently in CI compared to what we expect. Locally, running `pre-commit` without any args will only check _staged_ files. So I think in the CI, there are no _staged_ files, just files. So we might need `--all-files` to force checking of all files.

Reply:
Hmm, but we don't do that on others and it picks them up fine. And we don't want to run on all files because the next person has to fix _everything_ (which should not be a barrier to adding their changes).

Reply:
Seems like the solution is mentioned here: https://pre-commit.com/#usage-in-continuous-integration

We should use `pre-commit run --from-ref origin/HEAD --to-ref HEAD` to only check changed files in CI.

Reply:
Good spotting! Will have to add that into operator-templating at some point.